### DISCUSSION TITLE
Syslog Dashboard enrichement

### DISCUSSION THREAD

**Discussion by sleepingbel** [Security-Onion-Solutions/securityonion#2238] [Ideas]
hello,

Can there be created a list whith hostnames that are forwarding logs?

This is useful if you need to search a specific host's logs for a specific moment.

On this moment I am using syslog only for routers, switches and printers. I can not change the syslog output of these devices. 
On the syslog dashbord, none of the sub-screens have data .
![afbeelding](https://user-images.githubusercontent.com/29328171/101285331-cd37f600-37e4-11eb-9847-a0a1358aa4d1.png)

Regards
Bart

**Comment by weslambert** [comment]
I would assume the syslog is not being parsed as expected.  Are you able to locate the data by searching within Discover?  As far as the list of devices you mentioned, you can create your own custom dashboards in Kibana, using fields from the events.

**Reply by sleepingbel** [reply]
Hello Wes,

I can see the data in discover. I will post some of them
![afbeelding](https://user-images.githubusercontent.com/29328171/101381530-24a69680-38b7-11eb-9218-4115fa71ac25.png)
![Knipsel](https://user-images.githubusercontent.com/29328171/101381918-a26aa200-38b7-11eb-9cb0-45a40f1888bf.PNG)
![afbeelding](https://user-images.githubusercontent.com/29328171/101382087-d776f480-38b7-11eb-8414-77d5be88c55d.png)

the number between <> is the syslog priority
The first field is the timestamp
The second field is the hostname
And the rest is the message


And the log.source.address is recorded as IP address and port
![Knipsel](https://user-images.githubusercontent.com/29328171/101382761-9f23e600-38b8-11eb-8f22-80f6edf89910.PNG)

When I look at the parser it normally parses the hostname, right?
I checked these two,

[https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/syslog](url)
[https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/zeek.syslog](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/zeek.syslog)

I'll be honest, I've never written a parser before, but I'm now looking for all the information about it, so I can start with it.

I have tried to create a grok for the syslog of the router 
![Knipsel](https://user-images.githubusercontent.com/29328171/101381918-a26aa200-38b7-11eb-9cb0-45a40f1888bf.PNG)

`<%{INT:syslog.Priority}>%{SYSLOGTIMESTAMP:syslog.Timestamp} %{SYSLOGHOST:syslog.Host} %{SYSLOGPROG:syslog.Program} %{GREEDYDATA:syslog.Action} (%{GREEDYDATA:syslog.Src_IP} %{GREEDYDATA:syslog.Dst_IP} %{GREEDYDATA:syslog.Protocol})"`

The result is not yet perfect, but i found the developers tools in the elastic features

`{
  "syslog": {
    "Action": "Drop ",
    "Program": "ACL",
    "Dst_IP": "Dst=2.17.198.9",
    "Src_IP": "Src=192.168.1.106",
    "Priority": "4",
    "Host": "RV130W",
    "Protocol": "Protocol=ICMP",
    "Timestamp": "Dec  8 19:06:41"
  },
  "program": "ACL"
}`

I am now trying to remove the Src =, Dst = and Protocol = from the result. But no luck till now

Regards 
Bart

**Comment by abesinger** [comment]
I don't know if it's of any help, but you might want to take a look at the ingest pipeline I posted in #5251
